src/main/java/duke/DukeList.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+public class DukeList {
+    private ArrayList<Task> tasks;
+
+    public DukeList() {
+        tasks = new ArrayList();
+    }
+
+    /**
+     * Creates a new task and adds it to the list.
+     * @param task is the name of task to be added to the list.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Creates an array of task names in our list.
+     * The first element is a string that serves as a header for the list.
+     * Each subsequent element begins with the serial number which is 1 more than the task index.
+     * Each subsequent element of the array also has a status field.
+     * Y indicates that task is done and N indicates that the task is not done.
+     * @return Array of Strings that enumerates the tasks in our list. Null if list empty.
+     */
+    public String[] listTasks() throws DukeException {
+        if (Task.getTaskCount() > 0) {
+            String[] result = new String[Task.getTaskCount() + 1];
+
+            result[0] = "The following items are tracked:";
+            for (int i = 1; i <= Task.getTaskCount(); ++i) {
+                Task task = tasks.get(i - 1);
+                // Format list entry as <serial number>. [<status>] <task name>
+                result[i] = i + ". " + task.toString();
+            }
+            return result;
+        } else {
+            throw new DukeException("There are no tasks to list.");
+        }

```

Comment:
To improve code readability, make the happy path unindented as much as possible. It is the ‘unusual’ cases that should be indented.